openshift/message-board/message-service/src/main/java/org/hibernate/demo/message/post/core/service/procedure/RemoteProcedureRegister.java

```diff
@@ -0,0 +1,26 @@
+package org.hibernate.demo.message.post.core.service.procedure;
+
+import javax.ejb.Singleton;
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.StoredProcedureQuery;
+
+@Singleton
+public class RemoteProcedureRegister {
```

Comment:
I'm not sure what's going on here.
Why do you need to keep track if the storedprocedure has been initialized already?

Reply:
Marshallers from what I know could be registered even more than once. That is just an optimization for avoid useless invocations.